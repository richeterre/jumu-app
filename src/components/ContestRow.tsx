import React from "react";
import { TouchableOpacity, Text, StyleSheet } from "react-native";
import { flag } from "../helpers/countries";

interface Props {
  name: string;
  countryCode: string;
  onPress: () => void;
}

const ContestRow: React.FC<Props> = ({ name, countryCode, onPress }) => (
  <TouchableOpacity style={styles.root} onPress={onPress}>
    <Text>{`${flag(countryCode)} ${name}`}</Text>
  </TouchableOpacity>
);

const styles = StyleSheet.create({
  root: {
    padding: 15
  }
});

export default ContestRow;
