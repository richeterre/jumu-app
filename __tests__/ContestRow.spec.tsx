import React from "react";
import renderer from "react-test-renderer";

import ContestRow from "../src/components/ContestRow";

describe("ContestRow", () => {
  test("renders correctly", () => {
    const contestRow = renderer.create(
      <ContestRow
        contest={{
          id: "1",
          name: "RW Helsinki 2019",
          host: { id: "1", countryCodes: ["FI"] },
          dates: ["2019-01-01"],
          stages: [{ id: "1", name: "Aula" }],
        }}
        onPress={() => undefined}
      />
    );
    expect(contestRow).toMatchSnapshot();
  });
});
