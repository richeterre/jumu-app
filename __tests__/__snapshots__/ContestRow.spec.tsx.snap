// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContestRow renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "padding": 16,
    }
  }
>
  <Text>
    ðŸ‡«ðŸ‡® RW 2019, DS Helsinki
  </Text>
</View>
`;
